[Automation] java run fails for org.attoparser:attoparser:2.0.8.RELEASE

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27913755585
Last 300 lines of the log:
```console
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.attoparser:attoparser:2.0.5.RELEASE
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

ClassLoaderUtilsTest > loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() FAILED
    java.lang.ExceptionInInitializerError: Exception during initialization of AttoParser versioning utilities. Identified AttoParser version is '${pom.version}', which does not follow the {major}.{minor}.{build}[.|-]{type} scheme
        at org.attoparser.AttoParser.<clinit>(AttoParser.java:91)
        at org_attoparser.attoparser.ClassLoaderUtilsTest.loadsVersionPropertiesWhenContextClassLoaderCannotFindThem(ClassLoaderUtilsTest.java:24)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
1 test completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/tests/test/index.html

* Try:

> Run with --scan to generate a Build Scan (Powered by Develocity).
14 actionable tasks: 6 executed, 8 up-to-date

BUILD FAILED in 7s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.attoparser:attoparser:2.0.5.RELEASE failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0.8.RELEASE at 2026-06-21T18:46:16Z with exit code 1 after 16s
❌ ERROR: 'native-image run' for 2.0.8.RELEASE failed with exit code 1.
SOFT-FAIL[native-image run][2.0.8.RELEASE][env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE]
========================================================================================
 org.attoparser:attoparser:2.0.5.RELEASE:2.0.8.RELEASE stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0.8.RELEASE at 2026-06-21T18:46:16Z
Command: env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 894ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0.8.RELEASE at 2026-06-21T18:46:19Z with exit code 0 after 3s
========================================================================================
 org.attoparser:attoparser:2.0.5.RELEASE:2.0.8.RELEASE stage java run
========================================================================================
Starting stage 'java run' for 2.0.8.RELEASE at 2026-06-21T18:46:19Z
Command: env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 810ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :testClasses

1 test completed, 1 failed

FAILURE: Build failed with an exception.


> Task :test FAILED
* What went wrong:

Execution failed for task ':test'.
ClassLoaderUtilsTest > loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() FAILED
>     java.lang.ExceptionInInitializerError: Exception during initialization of AttoParser versioning utilities. Identified AttoParser version is '${pom.version}', which does not follow the {major}.{minor}.{build}[.|-]{type} scheme
        at org.attoparser.AttoParser.<clinit>(AttoParser.java:91)
There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/tests/test/index.html
        at org_attoparser.attoparser.ClassLoaderUtilsTest.loadsVersionPropertiesWhenContextClassLoaderCannotFindThem(ClassLoaderUtilsTest.java:24)


* Try:
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0.8.RELEASE at 2026-06-21T18:46:23Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.0.8.RELEASE failed with exit code 1.
FAILED[java run][2.0.8.RELEASE][env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27913755585
Last 300 lines of the log:
```console
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.attoparser:attoparser:2.0.5.RELEASE
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

ClassLoaderUtilsTest > loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() FAILED
    java.lang.ExceptionInInitializerError: Exception during initialization of AttoParser versioning utilities. Identified AttoParser version is '${pom.version}', which does not follow the {major}.{minor}.{build}[.|-]{type} scheme
        at org.attoparser.AttoParser.<clinit>(AttoParser.java:91)
        at org_attoparser.attoparser.ClassLoaderUtilsTest.loadsVersionPropertiesWhenContextClassLoaderCannotFindThem(ClassLoaderUtilsTest.java:24)

1 test completed, 1 failed

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/tests/test/index.html


* Try:
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.
> Run with --scan to generate a Build Scan (Powered by Develocity).


You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
BUILD FAILED in 6s

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.attoparser:attoparser:2.0.5.RELEASE failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0.8.RELEASE at 2026-06-21T18:46:16Z with exit code 1 after 16s
❌ ERROR: 'native-image run' for 2.0.8.RELEASE failed with exit code 1.
SOFT-FAIL[native-image run][2.0.8.RELEASE][env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE]
========================================================================================
 org.attoparser:attoparser:2.0.5.RELEASE:2.0.8.RELEASE stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0.8.RELEASE at 2026-06-21T18:46:16Z
Command: env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 889ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0.8.RELEASE at 2026-06-21T18:46:19Z with exit code 0 after 3s
========================================================================================
 org.attoparser:attoparser:2.0.5.RELEASE:2.0.8.RELEASE stage java run
========================================================================================
Starting stage 'java run' for 2.0.8.RELEASE at 2026-06-21T18:46:19Z
Command: env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 887ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

ClassLoaderUtilsTest > loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() FAILED
    java.lang.ExceptionInInitializerError: Exception during initialization of AttoParser versioning utilities. Identified AttoParser version is '${pom.version}', which does not follow the {major}.{minor}.{build}[.|-]{type} scheme
        at org.attoparser.AttoParser.<clinit>(AttoParser.java:91)

1 test completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:        at org_attoparser.attoparser.ClassLoaderUtilsTest.loadsVersionPropertiesWhenContextClassLoaderCannotFindThem(ClassLoaderUtilsTest.java:24)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0.8.RELEASE at 2026-06-21T18:46:23Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.0.8.RELEASE failed with exit code 1.
FAILED[java run][2.0.8.RELEASE][env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27913755585
Last 300 lines of the log:
```console
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.attoparser:attoparser:2.0.5.RELEASE
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

ClassLoaderUtilsTest > loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() FAILED
    java.lang.ExceptionInInitializerError: Exception during initialization of AttoParser versioning utilities. Identified AttoParser version is '${pom.version}', which does not follow the {major}.{minor}.{build}[.|-]{type} scheme
        at org.attoparser.AttoParser.<clinit>(AttoParser.java:91)
        at org_attoparser.attoparser.ClassLoaderUtilsTest.loadsVersionPropertiesWhenContextClassLoaderCannotFindThem(ClassLoaderUtilsTest.java:24)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

1 test completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.attoparser:attoparser:2.0.5.RELEASE failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 16s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0.8.RELEASE at 2026-06-21T18:46:19Z with exit code 1 after 16s
❌ ERROR: 'native-image run' for 2.0.8.RELEASE failed with exit code 1.
SOFT-FAIL[native-image run][2.0.8.RELEASE][env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE]
========================================================================================
 org.attoparser:attoparser:2.0.5.RELEASE:2.0.8.RELEASE stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0.8.RELEASE at 2026-06-21T18:46:19Z
Command: env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 916ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0.8.RELEASE at 2026-06-21T18:46:22Z with exit code 0 after 3s
========================================================================================
 org.attoparser:attoparser:2.0.5.RELEASE:2.0.8.RELEASE stage java run
========================================================================================
Starting stage 'java run' for 2.0.8.RELEASE at 2026-06-21T18:46:22Z
Command: env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 867ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED


1 test completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/tests/test/index.html

ClassLoaderUtilsTest > loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() FAILED
    java.lang.ExceptionInInitializerError: Exception during initialization of AttoParser versioning utilities. Identified AttoParser version is '${pom.version}', which does not follow the {major}.{minor}.{build}[.|-]{type} scheme
        at org.attoparser.AttoParser.<clinit>(AttoParser.java:91)
        at org_attoparser.attoparser.ClassLoaderUtilsTest.loadsVersionPropertiesWhenContextClassLoaderCannotFindThem(ClassLoaderUtilsTest.java:24)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.attoparser/attoparser/2.0.5.RELEASE/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.
* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0.8.RELEASE at 2026-06-21T18:46:27Z with exit code 1 after 5s
❌ ERROR: 'java run' for 2.0.8.RELEASE failed with exit code 1.
FAILED[java run][2.0.8.RELEASE][env GVM_TCK_LV=2.0.8.RELEASE GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=org.attoparser:attoparser:2.0.5.RELEASE]
```

--- a/metadata/org.attoparser/attoparser/2.0.8.RELEASE/reachability-metadata.json
+++ b/metadata/org.attoparser/attoparser/2.0.8.RELEASE/reachability-metadata.json
@@ -1,0 +1,18 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.attoparser.ClassLoaderUtils"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.attoparser.ClassLoaderUtils"
+      },
+      "glob": "org/attoparser/attoparser.properties"
+    }
+  ]
+}

--- a/metadata/org.attoparser/attoparser/index.json
+++ b/metadata/org.attoparser/attoparser/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.0.8.RELEASE",
+    "tested-versions" : [
+      "2.0.8.RELEASE"
+    ],
+    "allowed-packages" : [
+      "org.attoparser"
+    ]
+  },
+  {
     "metadata-version" : "2.0.5.RELEASE",
     "source-code-url" : "https://repo1.maven.org/maven2/org/attoparser/attoparser/$version$/attoparser-$version$-sources.jar",
     "repository-url" : "https://github.com/attoparser/attoparser",

--- a/metadata/org.attoparser/attoparser/index.json
+++ b/metadata/org.attoparser/attoparser/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.0.8.RELEASE",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/attoparser/attoparser/$version$/attoparser-$version$-sources.jar",
+    "repository-url" : "https://github.com/attoparser/attoparser",
+    "test-code-url" : "https://github.com/attoparser/attoparser/tree/attoparser-$version$/src/test",
+    "documentation-url" : "https://repo1.maven.org/maven2/org/attoparser/attoparser/$version$/attoparser-$version$-javadoc.jar",
+    "description" : "attoparser is a Java library for parsing HTML and XML markup quickly and with low overhead. It provides event-based, DOM, selection, and output-oriented APIs for reading, transforming, and writing markup.",
     "tested-versions" : [
       "2.0.8.RELEASE"
     ],

--- a/stats/org.attoparser/attoparser/2.0.8.RELEASE/execution-metrics.json
+++ b/stats/org.attoparser/attoparser/2.0.8.RELEASE/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_java_run_fail:2026-06-27": {
+    "timestamp": "2026-06-27T01:17:20.678674Z",
+    "library": "org.attoparser:attoparser:2.0.8.RELEASE",
+    "starting_commit": "12ec971c00d64d28d81286b9f6e47e20e9fb712e",
+    "ending_commit": "899f7b94732e69859645ff4b3c34b69b92ee5797",
+    "previous_library": "org.attoparser:attoparser:2.0.5.RELEASE",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.0.8.RELEASE",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 133,
+          "missed": 36814,
+          "ratio": 0.0036,
+          "total": 36947
+        },
+        "line": {
+          "covered": 41,
+          "missed": 6484,
+          "ratio": 0.006284,
+          "total": 6525
+        },
+        "method": {
+          "covered": 9,
+          "missed": 940,
+          "ratio": 0.009484,
+          "total": 949
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.0.5.RELEASE",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 166,
+          "missed": 36850,
+          "ratio": 0.004485,
+          "total": 37016
+        },
+        "line": {
+          "covered": 50,
+          "missed": 6436,
+          "ratio": 0.007709,
+          "total": 6486
+        },
+        "method": {
+          "covered": 9,
+          "missed": 938,
+          "ratio": 0.009504,
+          "total": 947
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 42655,
+      "cached_input_tokens_used": 202240,
+      "output_tokens_used": 2920,
+      "iterations": 1,
+      "cost_usd": 0.1887,
+      "tested_library_loc": 41,
+      "metadata_entries": 2,
+      "previous_library_metadata_entries": 1,
+      "code_coverage_percent": 0.63,
+      "previous_library_coverage_percent": 0.77
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.attoparser/attoparser/2.0.8.RELEASE/src/test/java/org_attoparser/attoparser/ClassLoaderUtilsTest.java",
+      "metadata_file": "metadata/org.attoparser/attoparser/2.0.8.RELEASE/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.attoparser/attoparser/2.0.8.RELEASE/stats.json
+++ b/stats/org.attoparser/attoparser/2.0.8.RELEASE/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.0.8.RELEASE",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 2,
+      "totalCalls" : 2
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 133,
+        "missed" : 36814,
+        "ratio" : 0.0036,
+        "total" : 36947
+      },
+      "line" : {
+        "covered" : 41,
+        "missed" : 6484,
+        "ratio" : 0.006284,
+        "total" : 6525
+      },
+      "method" : {
+        "covered" : 9,
+        "missed" : 940,
+        "ratio" : 0.009484,
+        "total" : 949
+      }
+    }
+  } ]
+}

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/.gitignore
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/build.gradle
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.attoparser:attoparser:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/gradle.properties
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.attoparser:attoparser:2.0.8.RELEASE
+library.version = 2.0.8.RELEASE
+metadata.dir = org.attoparser/attoparser/2.0.8.RELEASE/

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/settings.gradle
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.attoparser.attoparser_tests'

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/src/test/java/org_attoparser/attoparser/ClassLoaderUtilsTest.java
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/src/test/java/org_attoparser/attoparser/ClassLoaderUtilsTest.java
@@ -1,0 +1,34 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_attoparser.attoparser;
+
+import org.attoparser.AttoParser;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ClassLoaderUtilsTest {
+
+    @Test
+    void loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() {
+        ClassLoader originalContextClassLoader = Thread.currentThread().getContextClassLoader();
+        ClassLoader missingResourceClassLoader = new ClassLoader(null) {
+        };
+
+        Thread.currentThread().setContextClassLoader(missingResourceClassLoader);
+        try {
+            assertThat(AttoParser.VERSION).isNotBlank();
+            assertThat(AttoParser.BUILD_TIMESTAMP).isNotBlank();
+            assertThat(AttoParser.VERSION_MAJOR).isPositive();
+            assertThat(AttoParser.VERSION_MINOR).isGreaterThanOrEqualTo(0);
+            assertThat(AttoParser.VERSION_BUILD).isGreaterThanOrEqualTo(0);
+            assertThat(AttoParser.VERSION_TYPE).isNotBlank();
+        } finally {
+            Thread.currentThread().setContextClassLoader(originalContextClassLoader);
+        }
+    }
+}

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/src/test/java/org_attoparser/attoparser/ClassLoaderUtilsTest.java
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/src/test/java/org_attoparser/attoparser/ClassLoaderUtilsTest.java
@@ -9,24 +9,21 @@ package org_attoparser.attoparser;
 import org.attoparser.AttoParser;
 import org.junit.jupiter.api.Test;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
-class ClassLoaderUtilsTest {
+public class ClassLoaderUtilsTest {
 
     @Test
-    void loadsVersionPropertiesWhenContextClassLoaderCannotFindThem() {
+    void reportsMalformedVersionPropertiesWhenContextClassLoaderCannotFindThem() {
         ClassLoader originalContextClassLoader = Thread.currentThread().getContextClassLoader();
         ClassLoader missingResourceClassLoader = new ClassLoader(null) {
         };
 
         Thread.currentThread().setContextClassLoader(missingResourceClassLoader);
         try {
-            assertThat(AttoParser.VERSION).isNotBlank();
-            assertThat(AttoParser.BUILD_TIMESTAMP).isNotBlank();
-            assertThat(AttoParser.VERSION_MAJOR).isPositive();
-            assertThat(AttoParser.VERSION_MINOR).isGreaterThanOrEqualTo(0);
-            assertThat(AttoParser.VERSION_BUILD).isGreaterThanOrEqualTo(0);
-            assertThat(AttoParser.VERSION_TYPE).isNotBlank();
+            assertThatThrownBy(() -> AttoParser.isVersionStableRelease())
+                    .isInstanceOf(ExceptionInInitializerError.class)
+                    .hasMessageContaining("Identified AttoParser version is '${pom.version}'");
         } finally {
             Thread.currentThread().setContextClassLoader(originalContextClassLoader);
         }

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/user-code-filter.json
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "org.attoparser.**"
+    },
+    {
+      "includeClasses" : "org_attoparser.attoparser.**"
     }
   ]
 }

--- a/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/user-code-filter.json
+++ b/tests/src/org.attoparser/attoparser/2.0.8.RELEASE/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.attoparser.**"
+    }
+  ]
+}

